cmd/ci-reporter/report.go

```diff
@@ -38,167 +38,111 @@ type requiredJob struct {
 }
 
 func main() {
-	boolPtr := flag.Bool("short", false, "a short report for mails and slack")
+	// parse flags
+	isShortFlagSet := flag.Bool("short", false, "A short report for mails and slack")
+	releaseVersion := flag.String("v", "", "Adds specific K8s release version to the report like 1.22")
 	flag.Parse()
 
+	// set environment variables
 	githubAPIToken := os.Getenv("GITHUB_AUTH_TOKEN")
 	if githubAPIToken == "" {
 		fmt.Printf("Please provide GITHUB_AUTH_TOKEN env variable to be able to pull cards from the github board")
 		os.Exit(1)
 	}
 
-	releaseVersion := os.Getenv("RELEASE_VERSION")
+	// create a new github client
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: githubAPIToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
 
-	err := printCardsOverview(githubAPIToken, *boolPtr)
+	// GitHub Report
+	err := printGithubIssueReport(githubAPIToken, client, *isShortFlagSet)
 	if err != nil {
 		fmt.Printf("error when querying cards overview, exiting: %v\n", err)
 		os.Exit(1)
 	}
 
-	printJobsStatistics(releaseVersion)
+	// Testgrid Report
+	printTestgridReport(*releaseVersion)
 }
 
+// This regex is getting used to identify sig lables on github issues
+var sigRegex = regexp.MustCompile(`sig/[a-zA-Z-]+`)
+
+type (
+	columnTitle string
+	columnID    int64
+)
+
 const (
-	newCards                = 4212817
-	underInvestigationCards = 4212819
-	observingCards          = 4212821
-	ciSignalBoardProjectID  = 2093513
+	newColumn                columnID = 4212817
+	underInvestigationColumn columnID = 4212819
+	observingColumn          columnID = 4212821
+	resolvedColumn           columnID = 6798858
+
+	newColumnTitle       columnTitle = "New/Not Yet Started"
+	inFLightColumnTitle  columnTitle = "In flight"
+	observingColumnTitle columnTitle = "Observing"
+	resolvedColumnTitle  columnTitle = "Resolved"
 )
```

Comment:
The column information should preferably be stored a struct fields like:
```
type column struct {
  ID int
  Title string
} 
```

Reply:
Yes I will definitely do that in a future PR after we are clear if we would like to only scan the ci-signal project board.

I think it would be beneficial to scan all kubernetes issues (filter via lables) with this approach we would detect reported issues that are not on our radar yet.
(see: [api.github](https://docs.github.com/en/rest/reference/issues#list-repository-issues))
as done in the improvements (see: [leonardpahlke/ci-signal-report](https://github.com/leonardpahlke/ci-signal-report/blob/master/pkg/ci-reporter/github-reporter.go)